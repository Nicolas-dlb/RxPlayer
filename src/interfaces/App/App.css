.app {
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  background-color: #ececec;
}

.title {
  align-self: flex-start;
  font-weight: normal;
  margin-top: 10px !important;
  margin-bottom: 10px !important;
  margin-left: 15px !important;
}

/* Main page */
.hero-banner {
  display: flex;
  align-items: center;
  justify-content: center;
}
.hero-banner .btn {
  width: 100% !important;
}

@media only screen and (min-width: 999px) {
  .hero-banner {
    height: 50vh;
    width: 100%;
  }
}
