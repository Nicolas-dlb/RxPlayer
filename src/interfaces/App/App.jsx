import React, { useEffect, useState } from "react";
import "./App.css";
import Header from "../../components/header/Header";
import {
  getAllScenes,
  getBonusImages,
  getCrew,
  getSceneByTimecode,
} from "../../utils/api/api";

import Player from "../../components/player/Player";
import Scenes from "../Scenes/Scenes";
import Crew from "../Crew/Crew";
import BonusImages from "../BonusImages/BonusImages";
import Comments from "../Comments/Comments";

function App() {
  const [scenes, setScenes] = useState([]);
  const [crew, setCrew] = useState([]);
  const [bonusImages, setBonusImages] = useState([]);
  const [currentTimecode, setCurrentTimecode] = useState(0);
  const [reactions, setReactions] = useState([]);
  const [player, setPlayer] = useState([]);
  const [isDesktop, setDesktop] = useState(window.innerWidth > 999);

  const updateMedia = () => {
    if (isDesktop && window.innerWidth < 999) {
      setDesktop(false);
    } else if (!isDesktop && window.innerWidth > 999) {
      setDesktop(true);
    }
  };

  useEffect(() => {
    window.addEventListener("resize", updateMedia);
    return () => window.removeEventListener("resize", updateMedia);
  });

  useEffect(() => {
    getAllScenes().then((data) => setScenes(data));
    getCrew().then((data) => setCrew(data));
    getBonusImages().then((data) => setBonusImages(data));
  }, []);

  useEffect(() => {
    if (scenes.length > 0 && reactions.length === 0) {
      scenes.forEach((scene) => {
        if (scene.beginTimecode) {
          getSceneByTimecode(scene.beginTimecode).then((fetchedScene) => {
            if (fetchedScene.reactions) {
              setReactions((prev) =>
                isDesktop
                  ? [
                      ...fetchedScene.reactions.map((reaction) => ({
                        id: reaction.message + prev.length,
                        ...reaction,
                      })),
                      ...prev,
                    ]
                  : [
                      ...prev,
                      ...fetchedScene.reactions.map((reaction) => ({
                        id: reaction.message + prev.length,
                        ...reaction,
                      })),
                    ]
              );
            }
          });
        }
      });
    }
  }, [scenes]);

  return (
    <div className="app">
      <Header />
      <div id="test" className="hero-banner">
        <Player
          player={player}
          setPlayer={setPlayer}
          currentTimecode={currentTimecode}
          setCurrentTimecode={setCurrentTimecode}
        />

        {isDesktop && (
          <Comments
            reactions={reactions}
            setReactions={setReactions}
            currentTimecode={currentTimecode}
          />
        )}
      </div>
      <Scenes scenes={scenes} player={player} />
      <Crew crew={crew} />
      <BonusImages bonusImages={bonusImages} />
      {!isDesktop && (
        <Comments
          isDesktop={isDesktop}
          reactions={reactions}
          setReactions={setReactions}
          currentTimecode={currentTimecode}
        />
      )}
    </div>
  );
}

export default App;
