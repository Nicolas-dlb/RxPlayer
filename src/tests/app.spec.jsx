import { render } from "@testing-library/react";
import React from "react";
import App from "../interfaces/App/App";

describe("App", () => {
  it("Should render the app correctly", () => {
    const app = render(<App />);
    expect(app).toMatchSnapshot();
  });
});
