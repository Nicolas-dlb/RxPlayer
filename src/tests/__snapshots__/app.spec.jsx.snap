// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render the app correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <header>
          <div
            id="rxPlayer"
          >
            <a
              href="https://github.com/canalplus/rx-player"
            >
              <h1>
                <img
                  alt="RxPlayer"
                  src="rxPlayer_logo_white.png"
                />
              </h1>
            </a>
            <a
              class="version"
              href="https://github.com/canalplus/rx-player/releases"
            >
              <p>
                v3.27.0
              </p>
            </a>
          </div>
          <a
            aria-label="Go to Canal+ website"
            href="https://www.canalplus.com"
          >
            <img
              alt="CANAL+"
              id="canal"
              src="canalp.svg"
            />
          </a>
        </header>
        <div
          class="hero-banner"
          id="test"
        >
          <video
            controls=""
            id="video"
            preload="auto"
          />
          <div
            class="comments"
          >
            <button
              class="btn"
              type="button"
            >
              Add
            </button>
            <div
              class="comments-wrapper"
            />
          </div>
        </div>
        <div
          class="category"
          id="scenes"
        >
          <div
            class="category-header"
          >
            <h3
              class="title"
            >
              Scenes
            </h3>
          </div>
          <div
            class="category-content"
          />
        </div>
        <div
          class="category"
          id="crew"
        >
          <div
            class="category-header"
          >
            <h3
              class="title"
            >
              Crew
            </h3>
          </div>
          <div
            class="category-content"
          />
        </div>
        <div
          class="category"
          id="bonus-images"
        >
          <div
            class="category-header"
          >
            <h3
              class="title"
            >
              Bonus images
            </h3>
          </div>
          <div
            class="category-content"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <header>
        <div
          id="rxPlayer"
        >
          <a
            href="https://github.com/canalplus/rx-player"
          >
            <h1>
              <img
                alt="RxPlayer"
                src="rxPlayer_logo_white.png"
              />
            </h1>
          </a>
          <a
            class="version"
            href="https://github.com/canalplus/rx-player/releases"
          >
            <p>
              v3.27.0
            </p>
          </a>
        </div>
        <a
          aria-label="Go to Canal+ website"
          href="https://www.canalplus.com"
        >
          <img
            alt="CANAL+"
            id="canal"
            src="canalp.svg"
          />
        </a>
      </header>
      <div
        class="hero-banner"
        id="test"
      >
        <video
          controls=""
          id="video"
          preload="auto"
        />
        <div
          class="comments"
        >
          <button
            class="btn"
            type="button"
          >
            Add
          </button>
          <div
            class="comments-wrapper"
          />
        </div>
      </div>
      <div
        class="category"
        id="scenes"
      >
        <div
          class="category-header"
        >
          <h3
            class="title"
          >
            Scenes
          </h3>
        </div>
        <div
          class="category-content"
        />
      </div>
      <div
        class="category"
        id="crew"
      >
        <div
          class="category-header"
        >
          <h3
            class="title"
          >
            Crew
          </h3>
        </div>
        <div
          class="category-content"
        />
      </div>
      <div
        class="category"
        id="bonus-images"
      >
        <div
          class="category-header"
        >
          <h3
            class="title"
          >
            Bonus images
          </h3>
        </div>
        <div
          class="category-content"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
